### ISSUE TITLE
arctan2() returning incorrect value

### ISSUE THREAD

**Issue by RoseberryPi** [SciSharp/NumSharp#426]
So I'm using numsharp
`np.arctan2(np.array(-0.0012562886517319706), np.array(-0.7499033624114052))`

I get the wrong value, approximately, `-4.33e-5`

using numpy.net and C#'s Math library
`np.arctan2(np.array(-0.0012562886517319706), np.array(-0.7499033624114052))`
`Math.Atan2(-0.0012562886517319706, -0.7499033624114052)`

I get `-3.1399`, which is the correct value.. Am I just being dumb and doing something wrong or is NumSharp not actually calculating the correct value?

**Comment by mforezdev** [comment]
Hello @RoseberryPi,

You are right, looking at code I don't understand the type casting done to (byte*) instead of (double*).
There must be an explanation.

```csharp
case NPTypeCode.Double:
{
   var out_addr = (double*)out_y.Address;
   var out_addr_x = (byte*)out_x.Address;
   Parallel.For(0, len, i => *(out_addr)  = Converts.ToDouble(Math.Atan2(*(out_addr) + i, *(out_addr_x) + i)));
   return out_y;
}
```
I tried casting to (double*) and it returns the expected value, 3.13991738776297

**Comment by Nucs** [comment]
Good catch, and the diagnosis in the thread was spot on. That loop read the operands as (byte*) and reinterpreted raw bytes as doubles, so the output was never going to resemble an angle.

The arctan2 path has been rewritten since. It lands where you would expect now:

```
np.arctan2(np.array(-0.0012562886517319706), np.array(-0.7499033624114052));  // -3.1399173877629707
np.arctan2(np.array(1.0),  np.array(1.0));    // 0.7853981633974483   (45 degrees)
np.arctan2(np.array(1.0),  np.array(-1.0));   // 2.356194490192345    (135 degrees)
```

Bit-for-bit with NumPy 2.4.2. Thanks for the careful write-up, closing this out.
